packages/form-js-viewer/src/features/expression-language/variableExtractionHelpers.js

```diff
@@ -0,0 +1,167 @@
+import { parseExpressions, parseUnaryTests } from 'feelin';
+
+export const getFlavouredFeelVariableNames = (feelString, feelFlavour, options = {}) => {
+
+  const {
+    depth = 0,
+    specialDepthAccessors = {}
+  } = options;
+
+  if (![ 'expression', 'unaryTest' ].includes(feelFlavour)) return [];
+
+  const tree = feelFlavour === 'expression' ? parseExpressions(feelString) : parseUnaryTests(feelString);
+
+  const simpleExpressionTree = _buildSimpleFeelStructureTree(tree, feelString);
+
+  return (function _unfoldVariables(node) {
+
+    if (node.name === 'PathExpression') {
+
+      if (Object.keys(specialDepthAccessors).length === 0) {
+        return depth === 0 ? [ _getVariableNameAtPathIndex(node, 0) ] : [ ];
+      }
+
+      // if using special depth accessors, use a more complex extraction
+      return [ ..._smartExtractVariableNames(node, depth, specialDepthAccessors) ];
```

Comment:
Can't we just return `_smartExtractVariableNames(node, depth, specialDepthAccessors)`? Or was returning a reference a problem?

you could also do `[].concat(_smartExtractVariableNames(node, depth, specialDepthAccessors))` since it should be a bit faster (performance might not an issue in this case though)

Reply:
Oh actually wait it's because it's a set.